ext/liquid_c/vm.c

```diff
@@ -170,7 +171,7 @@ static void write_obj(VALUE output, VALUE obj)
 static inline void vm_stack_push(vm_t *vm, VALUE value)
 {
     VALUE *stack_ptr = (VALUE *)vm->stack.data_end;
-    assert(stack_ptr < (VALUE *)vm->stack.capacity_end);
+    assert(stack_ptr <= (VALUE *)vm->stack.capacity_end);
```

Comment:
We shouldn't be writing to `vm->stack.capacity_end`, since that would be past the end of the allocation.